gnucash/import-export/import-main-matcher.c

```diff
@@ -138,11 +145,45 @@ static void refresh_model_row (
                     GNCImportTransInfo *info);
 /* end local prototypes */
 
+static gboolean delete_hash (gpointer key, gpointer value, gpointer user_data)
+{
+    // Value is a hash table that needs to be destroyed. 
+    g_hash_table_destroy (value);
+    return TRUE;
+}
+
+static void
+commit_all (GNCImportMainMatcher *info)
+{
+    // Commit the edits for all accounts for which we've called inc_account_edit.
+    for (;info->edited_accounts; info->edited_accounts=info->edited_accounts->next)
+    {
+        int level = qof_instance_get_editlevel (info->edited_accounts->data);
+        g_assert(level == 1);
```

Comment:
Don't crash!  A SQL backend will already have written out the already-committed accounts so crashing will make a corrupted book. Instead log an error and then commit repeatedly until `editlevel == 0`.